src/ray/common/asio/instrumented_io_context.h

```diff
@@ -76,6 +76,63 @@ class instrumented_io_context : public boost::asio::io_context {
   void post(std::function<void()> handler, const std::string name = "UNKNOWN")
       LOCKS_EXCLUDED(mutex_);
 
+  /// Increments the  current and cumulative counts and returns the enqueueing time
+  /// for this handler.
+  /// The returned enqueueing time and handler stats pointer should be given to a
+  /// subsequent Instrument() call. The handler stats is also returned in order to
+  /// obviate the need for multiple handler stats table lookups, therefore reducing
+  /// contention on the table lock.
+  ///
+  /// \param name A human-readable name to which collected stats will be associated.
+  /// \return The time at which this handler was enqueued, in nanoseconds, and the
+  ///  handler stats.
+  std::pair<int64_t, std::shared_ptr<GuardedHandlerStats>> Enqueue(
+      const std::string name = "UNKNOWN");
```

Comment:
Can we make the name a required arg?

Reply:
Yeah good point, manual instrumentation should always require a human-readable handler name!